### DISCUSSION TITLE
search filter from svelte error 400

### DISCUSSION THREAD

**Discussion by bobwatcherx** [pocketbase/pocketbase#4248] [Q&A]
i see you docs in api preview. 
but whats wrong in code

 import PocketBase from 'pocketbase';

```
  let searchText = 'car beauty';
  let searchResults = [];

  const pb = new PocketBase('https://blalalala.pockethost.io');

  const fetchData = async () => {
    try {
      const records = await pb.collection('playerx').getFullList(
       {
         sort : '-created',
        filter : `title == ${searchText.value}`
       }
        );
      searchResults = records || [];
      console.log(searchResults)
    } catch (error) {
      console.error('Error fetching data:', error);
    }
  };

  onMount(() => {
    fetchData();
  });
```

**Comment by bobwatcherx** [comment]
```
Cari.svelte:22 Error fetching data: ClientResponseError 400: Invalid filter parameters.
    at pocketbase.js?v=4d0f0b12:883:13
    at async fetchData (Cari.svelte:13:20)
```

**Comment by ganigeorgiev** [comment]
1. There is no `==` operator, just `=`.
2. The variable must be wrapped in quotes (single or double). Alternatively, you can also use the [`pb.filter()`](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#binding-filter-parameters) helper to bind params from map:
    ```js
    filter: pb.filter("title={:title}", { title: searchText.value })
    ```

**Comment by bobwatcherx** [comment]
but why do I search for keywords that come up in an empty console log?

example
title on pocketbase
"japanese scenes"

but when I search and click search.
rather []

my code
```
<script>
  import { onMount } from 'svelte';
  import { Link } from 'svelte-navigator';
  import PocketBase from 'pocketbase';

  let searchText = 'japanese';
  let searchResults = [];

  const pb = new PocketBase('https://blalalaaa.pockethost.io');

 const fetchData = async () => {
  try {
    const records = await pb.collection('playerx').getFullList({
      sort: '-created',
      filter: pb.filter("title={:title}", { title: searchText})
    });
    searchResults = records || [];
    console.log(searchResults);
  } catch (error) {
    console.error('Error fetching data:', error);
  }
};


  onMount(() => {
    fetchData();
  });

 
</script>

<style>
  /* Tambahkan gaya CSS sesuai kebutuhan */
</style>

<div style="margin-top: 10px">
  <div class="container">
    <Link to="/" class="btn waves pink">Kembali</Link>
    <h5>Cari Bokep yg Lu Mau</h5>

    <input bind:value={searchText} placeholder="Search by title" />

    <button on:click={fetchData}>Search</button>

    {#if searchResults.length > 0}
      <ul>
        {#each searchResults as result (result.id)}
          <li>{result.title}</li>
        {/each}
      </ul>
    {:else}
      <p>No results found.</p>
    {/if}
  </div>
</div>
```

**Reply by ganigeorgiev** [reply]
Double check your collection API rules.

**Comment by bobwatcherx** [comment]
and give you fix . why this url parameter not work . from request api

https://blalaaaaaa.pockethost.io/api/collections/playerx/records?filter=(title='japanese')

**Reply by ganigeorgiev** [reply]
The query parameter must be properly URL encoded (in your case that would be `?filter=title%3D%27japanese%27`).

**Comment by bobwatcherx** [comment]
still not work sir
![Screenshot 2024-01-28 114435](https://github.com/pocketbase/pocketbase/assets/52072811/d945c3d9-0bbb-4a04-bb17-87f55e53abe6)
![Screenshot 2024-01-28 114415](https://github.com/pocketbase/pocketbase/assets/52072811/a546fd6d-5ac3-4bb6-a28b-65a1f93e96e3)

**Reply by ganigeorgiev** [reply]
You have a trailing `)` in the url.

**Reply by ganigeorgiev** [reply]
_Side-note: For more detailed errors you can check the request logs in the Admin UI > Logs._

**Comment by bobwatcherx** [comment]
my column table
![Screenshot 2024-01-28 114940](https://github.com/pocketbase/pocketbase/assets/52072811/663c556f-16f5-4f32-8809-8ffc41acd274)

**Comment by bobwatcherx** [comment]
![Screenshot 2024-05-05 142013](https://github.com/pocketbase/pocketbase/assets/52072811/c73d3f6d-6252-48a0-a79f-bf70caa08dd4)
![Screenshot 2024-05-05 142057](https://github.com/pocketbase/pocketbase/assets/52072811/a37ead6f-79e4-4c6d-afa3-15b14c264f25)

how to fix this

my code

```
export async function Cari_barang(teks) {
    try {
        await userAuth();
        const results = await pb.collection('hafidz_barang').getFullList({
            sort: '-created',
             filter: pb.filter({nama_barang : teks})
        });
        return results;
    } catch (error) {
        console.error("Error during searching for barang:", error);
        throw error;  
    }
}
```
I tried other functions like create , list , update, all of them work, only filtering based on fields doesn't work, what's the error?

**Comment by bobwatcherx** [comment]
because i want filter by nama_barang , or other field

**Comment by bobwatcherx** [comment]
i try this work 
` filter: pb.filter(`nama_barang='mycar'`)`

but not work if use variabel
` filter: pb.filter(`nama_barang={'${teks}'}`)`

how to fix it , can you add docs in pocketbase dashboard, i not find

**Reply by ganigeorgiev** [reply]
@bobwatcherx Please stop spamming. There is an example in the above comment - https://github.com/pocketbase/pocketbase/discussions/4248#discussioncomment-8268947.

I don't know how to make it more simple than this.
